app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -54,21 +54,22 @@
 
     @NonNull
     @Override
-    public Completable querySearch(@NonNull final String query) {
-        return freeSoundApiService.search(get(query))
-                                  .map(DefaultSearchDataModel::toResults)
-                                  .doOnSubscribe(__ -> searchStateOnceAndStream
-                                          .onNext(SearchState.inProgress()))
-                                  .doOnSuccess(this::storeValueAndClearError)
-                                  .doOnError(storeError(query))
-                                  .toCompletable()
-                                  .onErrorComplete();
+    public Completable querySearch(@NonNull final String query,
+                                   @NonNull final Completable afterThis) {
+        return afterThis.doOnSubscribe(__ -> reportInProgress())
```

Comment:
is there an easy way here to send a new state other than idle with the last result? Otherwise this will clear the previous search result and thus the listview instantly. Alternatively remember the last result in the SearchFragment until there are no searchresults